attributes/default.rb

```diff
@@ -10,17 +10,33 @@
 
 # 'binary' recipe
 
-default['bitcoin']['binary']['home'] = '/home/bitcoin'
+default['bitcoin']['binary']['home'] = "/home/#{node['bitcoin']['user']}"
 default['bitcoin']['binary']['version'] = '0.9.2.1'
 default['bitcoin']['binary']['checksum'] = '0060f7d38b98113ab912d4c184000291d7f026eaf77ca5830deec15059678f54'
 default['bitcoin']['binary']['service_wrapper'] = "bitcoind"
 default['bitcoin']['binary']['bitcoind'] = "bin/bitcoind"
+default['bitcoin']['source']['bitcoin-cli'] = node['bitcoin']['binary']['bitcoind']
 default['bitcoin']['binary']['nice_level'] = 10
 
 # 'package' recipe
 
 default['bitcoin']['package']['repo_url']['rhel'] = 'http://linux.ringingliberty.com/bitcoin/el7/x86_64/bitcoin-release-1-6.noarch.rpm'
 default['bitcoin']['package']['repo_checksum']['rhel'] = '1d3c3984d6155a504238834a27805ca93ab598a798c6bfd54625719d46aaa969'
-
 default['bitcoin']['package']['repo_url']['fedora'] = 'http://linux.ringingliberty.com/bitcoin/f20/x86_64/bitcoin-release-1-6.noarch.rpm'
 default['bitcoin']['package']['repo_checksum']['fedora'] = '11d6a2f28a3ae27ba040713a46ddb5b9553ea12aef7b442e9ae77a0469c3e90a'
+
+# 'source' recipe
+
+default['bitcoin']['source']['version'] = '0.10.1'
+default['bitcoin']['source']['checksum'] = '18a88785748588bb90764dc7f0aad5548da880d4d91f5803c7076cbcadb2959e'
+default['bitcoin']['source']['service_wrapper'] = "bitcoind"
+default['bitcoin']['source']['home'] = "/home/#{node['bitcoin']['user']}"
+default['bitcoin']['source']['prefix'] = "/usr/local"
+default['bitcoin']['source']['bitcoind'] = "#{node['bitcoin']['source']['prefix']}/bin/bitcoind"
+default['bitcoin']['source']['bitcoin-cli'] = "#{node['bitcoin']['source']['prefix']}/bin/bitcoin-cli"
+default['bitcoin']['source']['configure_options'] = "--disable-wallet --without-gui --prefix=#{node['bitcoin']['source']['prefix']}"
+default['bitcoin']['source']['make_options'] = "-j4"
```

Comment:
This will be counter-productive on single-core CPUs but since we don't have an easy way to detect the number of CPUs, I think it's fine to leave it like that and people could change it through their `run_list`.